File: 이진영/백준/1932.정수 삼각형.js
const input = require('fs').readFileSync('dev/stdin').toString().trim().split("\n");

const n = +input[0].trim();
const pyramid = input.slice(1).map(r => r.trim().split(" ").map(Number));

let dp = Array.from(Array(n), () => Array(n).fill(null));
dp[0][0] = pyramid[0][0];

for (let i = 1; i < n; i++) {
    for (let j = 0; j < i + 1; j++) {
        dp[i][j] = pyramid[i][j];
        if (j === 0) dp[i][j] += dp[i - 1][j];
        else if (dp[i - 1][j] === null) dp[i][j] += dp[i - 1][j - 1];
        else dp[i][j] += Math.max(dp[i - 1][j], dp[i - 1][j - 1])
    }
}

console.log(Math.max(...dp[n - 1]));

File: 정민선/백준/etc/1932_정수삼각형.py
n = int(input())
board = [[0] * n for _ in range(n)]
numbers = []
for _ in range(n):
    row = list(map(int, input().split()))
    if len(row) < n:
        row += [0] * (n - len(row))
    numbers.append(row)

board[0][0] = numbers[0][0]
for i in range(1, n):
    board[i][0] = board[i-1][0] + numbers[i][0]

for i in range(1, n):
    for j in range(1, n):
        board[i][j] = max(board[i-1][j] + numbers[i][j], board[i-1][j-1]+numbers[i][j])

print(max(board[-1]))